docs/design/crd-validation/README.md

```diff
@@ -0,0 +1,109 @@
+
+Some more details can be found [here](https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/).
+
+### In OCS Operator Project
+
+#### Goals
+
+Our main goals are,
+
+- Minimal changes to the existing OCS Operator project
+    - there should not be any (no or very minimal) change to the existing code
+- Easy to add / remove / modify validations
+    - other developers should be able to modify validations
+
+### Design
+
+To deploy an admission controller into OCS Operator project, we have to
+
+i. Generate TLS Certificate to authenticate the Admission Controller (AC)
+
+ii. Configure **WebHooks** to handle the incoming validation requests to AC
+
+iii. Configure OCS Operator CRDs to deploy AC pod
+
+Once **Admission Controller** is deployed to OCS Operator project, it will start
+
+    1. The server 
+    2. The service to this server, which are either Validating or Mutating Webhooks
+
+The webhook requires a valid TLS cerficate configured for a connection.
+
+WebHooks intercept object requests based on 3 parameters -
+
+1. The operation being performed for the request ( create, update, delete )
+2. The api group and version of the object (v1, v1beta)
+3. The type of resource ( pod, service, storagecluster) 
+
+
+#### ValidationWebhookConfiguration 
+
+Validating admission webhooks are invoked during the validation phase of the admission process. This phase allows the enforcement of invariants on particular API resources to ensure that the resource does not change again. The Pod Node Selector is also an example of a validation admission, by ensuring that all nodeSelector fields are constrained by the node selector restrictions on the project.
+
+Sample
+
+```
+apiVersion: admissionregistration.k8s.io/v1beta1
+  kind: ValidatingWebhookConfiguration  ----> 1
+  metadata:
+    name: <controller_name> ----> 2
+  webhooks:
+  - name: <webhook_name> ----> 3
+    clientConfig: ----> 4
+      service:
+        namespace: default  ----> 5
+        name: kubernetes ----> 6
+       path: <webhook_url> ----> 7
+      caBundle: <cert> ----> 8
+    rules: ----> 9
+    - operations: ----> 10
+      - <operation>
+      apiGroups:
+      - ""
+      apiVersions:
+      - "*"
+      resources:
+      - <resource>
+    failurePolicy: <policy> ----> 11
+```
+
+1. Specifies a validating admission webhook configuration.
+2. The name for the webhook admission object.
+3. The name of the webhook to call.
+4. Information about how to connect to, trust, and send data to the webhook server.
+5. The project where the front-end service is created.
+6. The name of the front-end service.
+7. The webhook URL used for admission requests.
+8. A PEM-encoded CA certificate that signs the server certificate used by the webhook server.
+9. Rules that define when the API server should use this controller.
+10. The operation that triggers the API server to call this controller are,
+
+    - create
+    - update
+    - delete
+    - connect
```

Comment:
This is a moderate amount of complexity. Why do we think creating this feature with this mechanism will be less error-prone than allowing the user to make their own changes?